tests/e2e/customsuites/directorypermissions.go

```diff
@@ -586,4 +587,323 @@ func (t *s3CSIDirectoryPermissionsTestSuite) DefineTests(driver storageframework
+	})
+
+	// --------------------------------------------------------------------
+	// 12. Pod umask MUST NOT override dir‑mode
+	//
+	// Mountpoint‑S3 applies the dir‑mode given at mount time and ignores
+	// the process umask when creating new directories.
+	//
+	// Test scenario:
+	//
+	//     [Pod] umask 077
+	//         │
+	//         ▼
+	//    mkdir /mnt/volume1/umask‑dir
+	//         │
+	//         ▼
+	//  [S3 Volume] (dir‑mode = 0755)
+	//
+	// Expectations:
+	// - mkdir succeeds inside the volume even with a restrictive umask
+	// - stat shows permissions 0755 (driver‑enforced), not 0700
+	// - UID/GID match the mount options (1001/2000 by default)
+	//
+	// This mirrors the file‑side umask test and proves the driver's
+	// directory‑permission logic is immune to container umask settings.
+	ginkgo.It("should override pod umask and honour dir‑mode", func(ctx context.Context) {
+		// Build volume with dir‑mode=0755
+		res := createVolume(ctx, testRegistry.config, pattern,
+			DefaultNonRootUser,  // uid
+			DefaultNonRootGroup, // gid
+			"0755")              // dir‑mode
+
+		// Pod runs as the same non‑root UID/GID
+		pod, err := CreatePodWithVolumeAndSecurity(ctx, f, res.Pvc, "",
+			DefaultNonRootUser, DefaultNonRootGroup)
+		framework.ExpectNoError(err)
+		defer e2epod.DeletePodWithWait(ctx, f.ClientSet, pod)
+
+		dir := "/mnt/volume1/umask-dir"
+
+		// Inside the container, set a *restrictive* umask then mkdir
+		_, _, err = e2evolume.PodExec(f, pod,
+			fmt.Sprintf(`sh -c 'umask 077; mkdir %s'`, dir))
+		framework.ExpectNoError(err, "mkdir with umask 077 failed")
+
+		// Verify the directory got the mount‑option mode (0755), not 0700
+		verifyDir(f, pod, dir, "755",
+			ptr.To(DefaultNonRootUser), ptr.To(DefaultNonRootGroup))
+	})
+
+	// --------------------------------------------------------------------
+	// 13. mv/rename MUST fail (directories)
+	//
+	// S3 has no native support for directory renames. Mountpoint‑S3 must
+	// reject mv/rename operations on directories, returning a clear error
+	// (e.g., EXDEV / EPERM / ENOTSUP).
+	//
+	// Test scenario:
+	//
+	//     mkdir /mnt/volume1/srcdir
+	//          │
+	//     mv /mnt/volume1/srcdir /mnt/volume1/dstdir
+	//          │
+	//          ▼
+	//     FAILS with:
+	//       - "Operation not permitted"
+	//       - "Operation not supported"
+	//       - "Function not implemented"
+	//
+	// Expectations:
+	// - mv/rename fails (exit code != 0)
+	// - Error message matches expected patterns (see above)
+	// - srcdir still exists (directory untouched)
+	// - dstdir does NOT exist (rename did not partially succeed)
+	// - Metadata (permissions, ownership) of srcdir unchanged
+	//
+	// Why this matters:
+	// This ensures the driver does not silently emulate rename via copy+delete
+	// or perform partial / broken renames, preserving S3's semantics and
+	// avoiding dangerous race conditions.
+	ginkgo.It("should ignore mkdir -m requested mode and apply dir‑mode", func(ctx context.Context) {
```

Comment:
The description above `"should ignore mkdir -m requested mode and apply dir‑mode"` matches the next test.

Reply:
Addressed